packages/vite/src/node/plugins/esbuild.ts

```diff
@@ -170,7 +172,7 @@ export async function transformWithEsbuild(
 
 export function esbuildPlugin(options: ESBuildOptions = {}): Plugin {
   const filter = createFilter(
-    options.include || /\.(tsx?|jsx)$/,
+    options.include || /\.[cm]?(tsx?|jsx)$/,
```

Comment:
We should probably revert this as they aren't `.ctsx`, `.mjsx` etc

Reply:
Updated.